src/Microsoft.VisualStudio.Composition/Configuration/CachedCatalog.cs

```diff
@@ -12,46 +12,29 @@ namespace Microsoft.VisualStudio.Composition
     using System.Text;
     using System.Threading;
     using System.Threading.Tasks;
+    using MessagePack;
+    using MessagePack.Resolvers;
     using Microsoft.VisualStudio.Composition.Reflection;
 
     public class CachedCatalog
     {
         protected static readonly Encoding TextEncoding = Encoding.UTF8;
 
-        public Task SaveAsync(ComposableCatalog catalog, Stream cacheStream, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task SaveAsync(ComposableCatalog catalog, Stream cacheStream, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
This is a breaking change because it will no longer be able to read the cache created by a prior version. That could be an issue for VS itself. We should think carefully about what the migration plan is. We _may_ want to leave these methods alone. 
And we probably want to expose whatever few MessagePack-related types would enable the app to serialize the objects itself, as will be necessary to allow object de-duping when serializing several catalogs in a row.